src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -554,12 +554,13 @@ CHIP_ERROR OTARequestor::SendQueryImageRequest(OperationalDeviceProxy & devicePr
 
     char location[DeviceLayer::ConfigurationManager::kMaxLocationLength];
     size_t codeLen = 0;
-    if ((DeviceLayer::ConfigurationMgr().GetCountryCode(location, sizeof(location), codeLen) == CHIP_NO_ERROR) && (codeLen > 0))
+    if ((DeviceLayer::ConfigurationMgr().GetCountryCode(location, sizeof(location), codeLen) == CHIP_NO_ERROR) && (codeLen == 2))
     {
         args.location.SetValue(CharSpan(location, codeLen));
     }
     else
     {
+        // Country code unavailable or invalid, use default
         args.location.SetValue(CharSpan("XX", strlen("XX")));
```

Comment:
Should we actually send anything in such a case given that the Location field is optional?

Reply:
Yes, that's probably more logical. As a last minute fix I think this is safer.